shanghai/network.py

```diff
@@ -211,7 +244,7 @@ def sendline(self, line: str):
         self.connection.writeline(line.encode(self.encoding))
 
     def sendcmd(self, command: str, *params: str):
-        args = [command, *params]
+        args = [command, *(str(p) for p in params)]
```

Comment:
That's kind of a violation of the signature though. I think it is sane to request all parameters to be strings.